### ISSUE TITLE
Working with visual studio

### ISSUE THREAD

**Issue by elyashiv** [exercism/c#451]
Currently there is no support for visual studio: we don't provide any way to compile with visual studio, and we don't check it in the CI.

But, it's a good IDE that people use, and it is mentioned in the getting started docs.

I had students that compiled in VS and didn't have warning on high level nor as errors, and this is a bad habit in c. Also, I have a hard time checking for platform specific problems if they use VS, because I need to setup a project.

I suggest one of two:
1. Moving to use a different build system (cmake, scons ...) instead of make, that supports VS
2. Adding vcproj files that students using VS can use.

I'm happy to help with this, but I will want to hear what other people think first.

**Comment by ryanplusplus** [comment]
The counter-point to this is why is VS special? Eclipse, VS Code, CLion, and others are all good IDEs that we could more explicitly support, but I'm personally in favor of taking a lowest-common denominator (`make`) approach that is IDE/editor agnostic. Is there not a `make`-based workflow that VS can use? I've not used an IDE that struggled with a `make`-based workflow.

**Comment by elyashiv** [comment]
I guess I didn't express my self right - it's not VS that is the problem, it's msbuild. You can compile nicely with the make file, as long as you use gcc or clang. msbuild just won't work - the setup in windows is different, the compile flags are different and so on.

So a student working on windows needs to install a linux enviroment (wsl, MinGW...). VS can probably work with this, but not easily. It works better with the compiler it's shipped with.

**Comment by elyashiv** [comment]
Currently in the [Installation guide](https://exercism.io/tracks/c/installation) file we instruct windows users to install VS (2017, I guess this needs an update). In the [Running the tests guide](https://exercism.io/tracks/c/tests) we tell the students to use WSL.

The instructions for VS specify just to add the files. No compiler options (warning level, warning -> errors) or memory check methods that are in the makefile for gcc.

Also, msbuild is a compiler that is used, and we can benefit from passing it through the CI.

**Comment by ryanplusplus** [comment]
Now I understand.

Does it make sense to just drop the VS instructions and recommend only WSL and MinGW? This would let us have a single makefile, single environment (unix-like), and single set of instructions for actually building and testing the code. If WSL and MinGW aren't great solutions, we could also explore recommending a cloud IDE.

**Comment by elyashiv** [comment]
I guess we can.

I think some mentors should answer this question - do they use a *nix system? Do they use WSL? Will they want better VS support?

**Comment by ryanplusplus** [comment]
I think the way we got to where we are is that none of the maintainers of the track were Windows users. If we have someone that is familiar with Windows and willing to add support then we can explore other options, but so far we haven't had that.

**Comment by patricksjackson** [comment]
I think there are a small enough number of people using this track on Visual Studio that not having the custom warning level and memchecks enabled is not really an issue, though that's just a guess based on not getting any questions about it. To be fair, maybe we'd have more users if there were better instructions, but I did test that the current instructions work.

With the v3 stuff I think they're also trying to move to providing a cloud editor and workspace (copying all the other coding sites), so this will cease to be a concern once that move is made.

I setup the C++ track to test on Windows (and Mac) using Github Actions, it might be worth doing that here as well.

**Comment by wolf99** [comment]
I use Windows. 

But not MSVS. C in VS is not fun. It is not a focus for MS. C99 has support only insofar as it supports C++ (C++ is the real focus), and MS are solid on not updating this for more recent versions of C
I can attest that WSL makes life much easier as allows for a "standard" GCC tool chain.

With those two reasons in mind, I would recommend that we do away with the suggestion of VS support entirely. We can then provide links to get setup with WSL instead and then refer to the Linux instructions for build chain setup. 

The only kicker here is that WSL is only available on Windows 10 1607. While that was released a ways back in August 2016 there are may be a number of people that may be on older Windows versions (no stats for this but IT media regularly tells us there is a cohort that don't want to upgrade 🤷) and Exercism in general does implicitly support older versions (See system reqs: https://github.com/exercism/windows-installer). Options for this include requiring a minimum OS version (2016 is some time ago now) for the track, or referring to MinGW or perhaps Cygwin.

Personally its a long long time since I have used MinGW and I have never used Cygwin.

Perhaps @iHiD may have browser stats to indicate the OS version people are using, what proportion of them are on Windows OS older than Windows 10 ? (even better if stats are specific to C track if possible). From there we could assess the need for support for that cohort.
